cirq/linalg/transformations.py

```diff
@@ -326,6 +324,14 @@ def partial_trace(tensor: np.ndarray, keep_indices: List[int]) -> np.ndarray:
     return np.einsum(tensor, left_indices + right_indices)
 
 
+class EntangledStateError(ValueError):
+    """Raised by `sub_state_error` if the result of factoring is not a pure state
```

Comment:
"not a pure state" -> "not a product state"

Also, I'd remove "by `sub_state_error`" since the exception may be used in future by related functions.

Suggestion:

```python
"""Raised when a product state is expected, but an entangled state is provided."""
```

Reply:
Changed to use your nice wording